src/newmatrix.cpp

```diff
@@ -60,6 +60,27 @@ void make_seen(struct matrix_icon *icon, int idnum)
   icon->decker->seen = seen;
 }
 
+void clear_hitcher(struct char_data *ch, bool shouldNotify)
+{
+    // Safety check: ensure ch is valid and actually hitched to someone
+    if (!ch || !ch->hitched_to) {
+        return;
+    }
+
+    // Notify the hitcher if requested
+    if (shouldNotify) {
+        send_to_char("Your hitcher has disconnected.\r\n", ch->hitched_to);
+    }
+
+    // Clear references
+    // Make sure ch->hitched_to->persona and ch->hitched_to->persona->decker are valid in your code
+    ch->hitched_to->persona->decker->hitcher = nullptr;
```

Comment:
Please check the full chain here (validate `persona` and `decker`) so that one of these unexpectedly unsetting doesn't crash the game.